src/config.c

```diff
@@ -3114,6 +3114,7 @@ standardConfig static_configs[] = {
     createBoolConfig("aof-disable-auto-gc", NULL, MODIFIABLE_CONFIG | HIDDEN_CONFIG, server.aof_disable_auto_gc, 0, NULL, updateAofAutoGCEnabled),
     createBoolConfig("replica-ignore-disk-write-errors", NULL, MODIFIABLE_CONFIG, server.repl_ignore_disk_write_error, 0, NULL, NULL),
     createBoolConfig("hide-user-data-from-log", NULL, MODIFIABLE_CONFIG, server.hide_user_data_from_log, 0, NULL, NULL),
+    createBoolConfig("allow-arbitrary-lazyexpires", NULL, MODIFIABLE_CONFIG | HIDDEN_CONFIG, server.allow_arbitrary_lazyexpires, 1, NULL, NULL),
```

Comment:
maybe `allow` is better changed to `avoid` (since it doesn't completely prevent them).
also, the config name doesn't mention anything about nesting, i wonder if we can add something to hint this, and still get a rather short name.